raft/log_unstable.go

```diff
@@ -55,22 +55,16 @@ func (u *unstable) maybeLastIndex() (uint64, bool) {
 // is any.
 func (u *unstable) maybeTerm(i uint64) (uint64, bool) {
 	if i < u.offset {
-		if u.snapshot == nil {
-			return 0, false
-		}
-		if u.snapshot.Metadata.Index == i {
+		if u.snapshot != nil && u.snapshot.Metadata.Index == i {
 			return u.snapshot.Metadata.Term, true
 		}
 		return 0, false
 	}
 
-	last, ok := u.maybeLastIndex()
```

Comment:
can we revert this change? i think the old one is more readable.

Reply:
But, I think it's really a simple case, it's also very common to write such statements in go.

Anyhow, if you want it to be changed (just mention), I will change it.

Thank you.

Reply:
please revert this change. thanks!

Reply:
made necessary changes

Thank you.